x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -230,7 +230,7 @@ export function LensEditConfigurationFlyout({
       },
       references,
       visualizationType: visualization.activeId,
-      title: visualization.activeId ?? '',
+      title: (attributes.title || visualization.activeId) ?? '',
```

Comment:
I'm not sure what the original intent of this line was, as the `attributes.title` value spread above, cannot be undefined.

With this change, the `attributes.title` value cannot be undefined so at worst an empty title is replaced with the `visualization.activeId`. But if a user wants an empty title we would be overriding that. Maybe we don't override the title at all unless there is a good reason.

Thoughts???

Reply:
cc @stratoula I believe you have coded that part

Reply:
Isn't it empty on a by value visualization? Have you checked this scenario?

Reply:
The default by value `title` is set to `""`. But isn't it strange to create the by value vis without a title then only when editing the vis in the flyout we set the title to a default title?

Reply:
If I understood correctly the problem is not with "by value visualizations" but with "by reference visualizations".
To reproduce the issue you have to save your visualization into the library, then when you edit your visualization inline the title changes to that `lnsXY` value. (that is not a default title but something else).
Am I right?

Reply:
Also defaulting to `activeId` changed the savedObject name to `lnsXY`. Now if you take a look at the visualization library you will notice that that visualization is called `lnsXY`

Reply:
It actually applies to both by value and by reference visualizations.

Reply:
Ok I removed the override altogether so it does not apply to `""` titles. See 620063e